docs: clarify Vault/pgsodium root key lifecycle and portability (#47876)
supabase/supabase: apps/docs/content/guides/database/extensions/pgsodium.mdx

description: 'Encryption library for Postgres'
---

> [Supabase Vault](/docs/guides/database/vault) is the recommended replacement for `pgsodium` and should be used instead. Supabase **does not recommend** any new usage of [`pgsodium`](https://github.com/michelp/pgsodium).

Reviewer: I would probably add this message to a danger admonition.

So people's attention get caught and are encouraged to migrate.

```suggestion
Supabase **does not recommend** the usage of [pgsodium](https://github.com/michelp/pgsodium). Use [Supabase Vault](/docs/guides/database/vault) instead.
```

Author: Do we ever start any doc with an admonition though? :)

Reviewer: I don't see a problem in particular.

But I understand it could be odd. Let me do a re-review of the document to propose something instead.

Reviewer: Seeing the opening words. I feel this piece is missing a description of what the extension does and what it is used for, and that's were the problem resides in terms of content. So I would do something like this:

**<Introduction on what extension is and does>**

<Admonition type="caution">
Supabase **does not recommend** the usage of [pgsodium](https://github.com/michelp/pgsodium) as it will be deprecated. Use [Supabase Vault](/docs/guides/database/vault) instead.

We will reach out to owners of impacted projects to assist with migrations away from [`pgsodium`](https://github.com/michelp/pgsodium) once the deprecation process begins.

Vault and `pgsodium` are separate extensions. Vault doesn't depend on `pgsodium` and is **not** affected by this deprecation.
</Admonition>

**<Rest of the page>**

Author: Ok, edited.
